<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">


    
!<!--                           Gazebo                        -->

    <!-- Gazebo Entegrasyonu -->
    <gazebo>
    <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
        <parameters>$(find robodog)/config/robodog_controller.yaml</parameters>
    </plugin>	
    </gazebo>

    <!-- ROS2 Kontrol Tanımlamaları -->
    <ros2_control name="RobodogSystem" type="system">
    <hardware>
        <plugin>gazebo_ros2_control/GazeboSystem</plugin>
    </hardware>

        <xacro:macro name="joint_interface" params="joint_name">
        <joint name="${joint_name}">
            <command_interface name="position">
                <param name="min">${-pi}</param>
                <param name="max">${pi}</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_value">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </xacro:macro>

    <xacro:joint_interface joint_name="front_left_leg_P0_R0_joint"/>
    <xacro:joint_interface joint_name="front_left_leg_P1_R1_joint"/>
    <xacro:joint_interface joint_name="front_left_leg_P2_R2_joint"/>
    <xacro:joint_interface joint_name="front_right_leg_P0_R0_joint"/>
    <xacro:joint_interface joint_name="front_right_leg_P1_R1_joint"/>
    <xacro:joint_interface joint_name="front_right_leg_P2_R2_joint"/>
    <xacro:joint_interface joint_name="back_left_leg_P0_R0_joint"/>
    <xacro:joint_interface joint_name="back_left_leg_P1_R1_joint"/>
    <xacro:joint_interface joint_name="back_left_leg_P2_R2_joint"/>
    <xacro:joint_interface joint_name="back_right_leg_P0_R0_joint"/>
    <xacro:joint_interface joint_name="back_right_leg_P1_R1_joint"/>
    <xacro:joint_interface joint_name="back_right_leg_P2_R2_joint"/>

</ros2_control>
<!-- 
    <gazebo>
        <plugin name="gazebo_ros_joint_state_publisher"
            filename="libgazebo_ros_joint_state_publisher.so">
            <update_rate>10</update_rate>
            <joint_name>front_left_leg_P0_R0_joint</joint_name>
            <joint_name>front_left_leg_P1_R1_joint</joint_name>
            <joint_name>front_left_leg_P2_R2_joint</joint_name>
            <joint_name>front_right_leg_P0_R0_joint</joint_name>
            <joint_name>front_right_leg_P1_R1_joint</joint_name>
            <joint_name>front_right_leg_P2_R2_joint</joint_name>
            <joint_name>back_left_leg_P0_R0_joint</joint_name>
            <joint_name>back_left_leg_P1_R1_joint</joint_name>
            <joint_name>back_left_leg_P2_R2_joint</joint_name>
            <joint_name>back_right_leg_P0_R0_joint</joint_name>
            <joint_name>back_right_leg_P1_R1_joint</joint_name>
            <joint_name>back_right_leg_P2_R2_joint</joint_name>
        </plugin>
    </gazebo>
  -->


</robot>